atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java

```diff
@@ -1493,19 +1480,28 @@ CfDef getCfForTable(TableReference tableRef, byte[] rawMetadata, int gcGraceSeco
     private <T> ClosableIterator<RowResult<T>> getRangeWithPageCreator(
             TableReference tableRef,
             RangeRequest rangeRequest,
-            long timestamp,
+            long startTs,
             ConsistencyLevel consistency,
             Supplier<ResultsExtractor<T>> resultsExtractor) {
+        SlicePredicate predicate;
+        if (rangeRequest.getColumnNames().size() == 1) {
+            byte[] colName = rangeRequest.getColumnNames().iterator().next();
+            predicate = SlicePredicates.latestVersionForColumn(colName, startTs);
+        } else {
+            // TODO(nziebart): optimize fetching multiple columns by performing a parallel range request for
```

Comment:
Is this work going to happen in this PR, a separate PR, or to be tracked as an issue?

Reply:
created https://github.com/palantir/atlasdb/issues/2498